pkg/verifier/aws/aws_verifier.go

```diff
@@ -339,11 +340,30 @@ func (a *AwsVerifier) createEC2Instance(input createEC2InstanceInput) (string, e
 
 	// Wait up to 5 minutes for the instance to be running
 	waiter := ec2.NewInstanceRunningWaiter(a.AwsClient)
+
 	if err := waiter.Wait(input.ctx, &ec2.DescribeInstancesInput{InstanceIds: []string{instanceID}}, 2*time.Minute); err != nil {
+
+		waiterErr := fmt.Errorf("%s: terminated %s after timing out waiting for instance to be running", err, instanceID)
+
+		resp, err := a.AwsClient.DescribeInstances(context.TODO(), &ec2.DescribeInstancesInput{
+			InstanceIds: []string{instanceID},
+		})
+		if err != nil {
+			fmt.Println("Warning: Waiter Describe instances failure.")
+		}
+
+		stateCode := *resp.Reservations[0].Instances[0].StateReason.Code
+		stateMessage := *resp.Reservations[0].Instances[0].StateReason.Message
+
+		if stateCode == "Client.InvalidKMSKey.InvalidState" {
```

Comment:
We should throw this string in a constant

Reply:
Done